handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -306,6 +306,16 @@ public void operationComplete(Future<SslContext> future) throws Exception {
     }
 
     /**
+     * The default implementation will simply call {@link AsyncMapping#map(Object, Promise)} but
+     * users can override this method to implement custom behavior.
+     *
+     * @see AsyncMapping#map(Object, Promise)
+     */
+    protected Future<SslContext> lookup(ChannelHandlerContext ctx, String hostname) throws Exception {
```

Comment:
Having it `protected` could be useful for something like this:

```java
public class MyAsyncMapping implements AsyncMapping<String, SslContext> {

  @Override
  public Future<SslContext> map(String input, Promise<SslContext> promise) {
    throw new UnsupportedOperationException();
  }
  
  public Future<SslContext> lookup(ChannelHandlerContext ctx, String hostname) {
    // ... do something ...
  }
}

public class MySniHandler extends SniHandler {
  
  public MySniHandler(AsyncMapping<String, SslContext> mapper) {
    super(mapper);
  }
  
  @Override
  public Future<SslContext> lookup(ChannelHandlerContext ctx, String hostname) {
    return ((MyAsyncMapping)mapping).lookup(ctx, hostname);
  }
}
```

On the other hand, the user has the opportunity to get a hold onto the Mapper via ctor if they need to. I think having it protected in this case would save one field.